app/src/main/java/com/babylon/wallet/android/presentation/transaction/analysis/TransactionAnalysisDelegate.kt

```diff
@@ -60,104 +57,63 @@ class TransactionAnalysisDelegate @Inject constructor(
+            runCatching {
+                transactionRepository.analyzeTransaction(
+                    manifestData = manifestData,
+                    isWalletTransaction = _state.value.requestNonNull.isInternal,
+                    notaryPublicKey = _state.value.ephemeralNotaryPrivateKey.toPublicKey()
+                )
+            }.then { transactionToReview ->
+                _state.update { it.copy(transactionManifestData = TransactionManifestData.from(transactionToReview.transactionManifest)) }
+
+                val manifestSummary = transactionToReview.transactionManifest.summary
+                resolveNotaryAndSignersUseCase(
+                    accountsAddressesRequiringAuth = manifestSummary.addressesOfAccountsRequiringAuth,
+                    personaAddressesRequiringAuth = manifestSummary.addressesOfPersonasRequiringAuth,
+                    notary = _state.value.ephemeralNotaryPrivateKey
+                ).map { notaryAndSigners ->
+                    transactionToReview.executionSummary
+                        .resolvePreview(notaryAndSigners)
+                        .resolveFees(notaryAndSigners)
+                }
+            }.then { transactionFees ->
+                _state.update { it.copy(transactionFees = transactionFees) }
 
-        return transactionRepository.getTransactionPreview(
-            TransactionPreviewRequest(
-                manifest = manifestData.instructions,
-                startEpochInclusive = startEpochInclusive.toLong(),
-                endEpochExclusive = endEpochExclusive.toLong(),
-                tipPercentage = 0,
-                nonce = Nonce.secureRandom().value.toLong(),
-                signerPublicKeys = notaryAndSigners.signersPublicKeys().map { it.asGatewayPublicKey() },
-                flags = TransactionPreviewRequestFlags(
-                    useFreeCredit = true,
-                    assumeAllSignatureProofs = false,
-                    skipEpochCheck = false
-                ),
-                optIns = TransactionPreviewOptIns(
-                    radixEngineToolkitReceipt = true // must be true
-                ),
-                blobsHex = manifestData.blobs.map { it.hex },
-                notaryPublicKey = notaryAndSigners.notaryPublicKeyNew().asGatewayPublicKey(),
-                notaryIsSignatory = notaryAndSigners.notaryIsSignatory
-            )
-        ).fold(
-            onSuccess = { preview ->
-                if (preview.receipt.isFailed) {
-                    val errorMessage = preview.receipt.errorMessage.orEmpty()
-                    val isFailureDueToDepositRules = errorMessage.contains("AccountError(DepositIsDisallowed") ||
-                        errorMessage.contains("AccountError(NotAllBucketsCouldBeDeposited")
-                    if (isFailureDueToDepositRules) {
-                        Result.failure(RadixWalletException.PrepareTransactionException.ReceivingAccountDoesNotAllowDeposits)
```

Comment:
What about this kind of error?

Reply:
Good catch! It was moved to Sargon, but we still had to properly map it into our own type.